pinot-common/src/main/java/com/linkedin/pinot/common/data/Schema.java

```diff
@@ -419,31 +400,24 @@ public Schema build() {
   }
 
   @Override
-  public boolean equals(Object o) {
-    if (isSameReference(this, o)) {
+  public String toString() {
+    return getJSONSchema();
+  }
+
+  @Override
+  public boolean equals(Object anObject) {
+    if (this == anObject) {
       return true;
     }
-
-    if (isNullOrNotSameClass(this, o)) {
-      return false;
+    if (anObject instanceof Schema) {
```

Comment:
Actually it is null safe and it is how String implement its equals(). (null instanceOf AnyClass) always returns false.